### DISCUSSION TITLE
Does the Uno team have the ability to create a designer plugin for WinUI

### DISCUSSION THREAD

**Discussion by HaoLife** [unoplatform/uno#15977] [Q&A]
### What would you like to be added

I found during the development process using WinUI that without a designer to design pages, the development efficiency is very slow. Need to constantly compile, run, and modify. The hot loading mode is often not easy to use, it may not take effect periodically, it may not work when there are many modifications to xaml, and viewmodel modifications may not take effect, among other issues. Considering that Microsoft does not develop designers, can we create plugin mode for preview

### Why is this needed

The development efficiency of WinUI is too poor

**Comment by sasakrsmanovic** [comment]
I'd be very curious to hear more and why Hot Reload doesn't work for you and to see what your needs would be. If you have 15 min time to discuss over a Zoom call that be greatly appreciated. Just reach out at sasha-at-platform.uno please.

**Comment by HaoLife** [comment]
The issue of work efficiency, during the development process of xaml, if I have a designer, I can get what I see and constantly change xaml to achieve the desired results. However, hot loading requires a complete compilation of the entire project, and many contents cannot be adjusted during the modification process of xaml, such as viewmodel and global resources. And hot loading often does not update the effect, requiring recompilation before restarting. I have very low efficiency in developing uno, and xaml editors cannot see the results, let alone the problems caused by different platforms.

**Comment by HaoLife** [comment]
This is like email and phone, the designer is like email, and hot loading is like phone. I don't want hot loading to interrupt me during my development process.

What we are looking for may be an asynchronous running previewer, with logs reminding us of errors on the page, and we can immediately see the layout effect during the development process, rather than imagining what effect it is through our brains.

**Comment by HaoLife** [comment]
There is a significant difference in the hot loading experience between VS and front-end development.
1. During the runtime, I am unable to edit the entire project, such as adding packages to introduce new controls, adding new ViewModels, models, and Pages.
2. After the xaml error, the pop-up window kept interrupting my development.
3. When my page path is very deep, I need data to enter this module, and the simulation is not intuitive enough.
4. He can directly set different resolutions to view the layout
5. I don't need to compile because I can't do anything during compilation and can only wait.

I have worked as a front-end developer before, and I think the hot loading in front-end development is very useful. It can modify anything without too many restrictions, but VS is different. I always want to stop debugging.

I know some typescript and I am trying to develop HarmonyOS using DevEco Studio development tools. It gave me a very good experience. The preview program allows me to view the results while writing the page. I think what we need is such a tool, not an XAML designer (the split screen design of the designer will reduce the editing area for developers). I think you can give it a try when using their development tools.
